### PR TITLE
POC add graph to dashboard

### PR DESCRIPTION
**Pull request by snopoke**
Once https://github.com/dimagi/commcare-hq/pull/15314 gets merged we'll be able to filter the data to only show for a specific domain.

image

### CODE REVIEW

File: corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js [hunk 1]
```diff
@@ -151,6 +151,42 @@
...
+
+        self.init = function (attrs) {
+            $scope.title = attrs.title;
+            $scope.slug = attrs.slug;
+            self.refreshGraph();
+        };
+
+        self.refreshGraph = function () {
+            djangoRMI.update_tile({
+                slug: $scope.slug
```
Comment (dimagimon) [inline corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js [hunk 2]
```diff
@@ -255,9 +291,24 @@
...
+        var link = function ($scope, element, attrs, controller) {
+            controller.init(attrs);
+        };
+
+        return {
+            scope: true,
+            restrict: 'EA',
+            templateUrl: utils.getTemplate(dashboardConfig, 'datadog_tile.html'),
+            controller: dashboardControllers.DatadogTileController,
+            link: link
```
Comment (dimagimon) [inline corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/dashboard/models.py [hunk 3]
```diff
@@ -1,8 +1,15 @@
+import json
+from urllib import urlencode
```
Comment (dimagimon) [inline corehq/apps/dashboard/models.py]: F401 'urlencode' imported but unused

File: corehq/apps/dashboard/models.py [hunk 4]
```diff
@@ -171,6 +179,59 @@ def context(self):
...
+    @quickcache(['self.tile_config.slug'], timeout=3600)
+    def get_graph(self):
+        start, end = self.times
+        response = requests.get(
+            'https://app.datadoghq.com/api/v1/graph/snapshot',
+            params={
+                'metric_query': "sum:commcare.xform_submissions.201{environment:production}.as_rate()",
+                'graph_def': json.dumps({
+                    "requests": [{
+                      "q": "sum:commcare.xform_submissions.201{environment:production}.as_count()",
```
Comment (dimagimon) [inline corehq/apps/dashboard/models.py]: E121 continuation line under-indented for hanging indent

**Comment by snopoke** [comment]
@dimagi/product @dimagi/ux what do you think? Not sure about the placement of the graph (was just playing with the idea).

File: corehq/apps/dashboard/models.py [hunk 5]
```diff
@@ -171,6 +178,59 @@ def context(self):
...
+        start = end - datetime.timedelta(days=7)
+        return posix(start), posix(end)
+
+    @quickcache(['self.tile_config.slug'], timeout=3600)
+    def get_graph(self):
+        start, end = self.times
+        response = requests.get(
+            'https://app.datadoghq.com/api/v1/graph/snapshot',
+            params={
+                'metric_query': "sum:commcare.xform_submissions.201{environment:production}.as_rate()",
```
Comment (benrudolph) [inline corehq/apps/dashboard/models.py]: would have to use SERVER_ENVIRONMENT after POC

**Review by benrudolph — COMMENTED** [review]
Code looks good. Interesting idea. I'm also not positive about the placement of it. However this makes me think, could we replace admin reports with datadog?

File: corehq/apps/dashboard/models.py (hunk 5 above)
Comment (snopoke) [inline corehq/apps/dashboard/models.py]: yea, would need a combo of `SERVER_ENVIRONMENT` and `domain:{}`

**Comment by snopoke** [comment]
Not sure about replacing admin report - the top level ones maybe but we couldn't support drill down

**Comment by snopoke** [comment]
Re admin reports though - would love to be able to stop querying ES directly for those numbers - especially the forms and cases

**Comment by benrudolph** [comment]
That's true, wishful thinking. Wonder how often those drilldowns are used...

**Comment by biyeun** [comment]
@snopoke I would place it between Applications and Reports the top, so it spans two columns in the middle of the dashboard.

**Comment by orangejenny** [comment]
This is cool! Didn't look closely at code, but wondering if it would make sense to constrain which domains can see it (e.g., don't show for domains that don't have any submissions yet).

File: corehq/apps/dashboard/models.py [hunk 6]
```diff
@@ -214,7 +214,7 @@ def get_graph(self):
                 'metric_query': "sum:commcare.xform_submissions.201{environment:production}.as_rate()",
                 'graph_def': json.dumps({
                     "requests": [{
-                        "q": "sum:commcare.xform_submissions.201{{{}}}.as_count()".format(tags),
```
Comment (benrudolph) [inline corehq/apps/dashboard/models.py]: that's some fun `{`

**Review by benrudolph — APPROVED** [review]

**Comment by snopoke** [comment]
Updated:
* moved graph as suggested (see updated screenshot in PR description)
* put behind feature flag

**Comment by snopoke** [comment]
build timed out: :boot:

**Pull request merged**